### PR TITLE
Cumulus 3291 coverage minimums

### PR DESCRIPTION
**Pull request by etcart**
**Summary:** Summary of changes

incorporate coverage tests into bamboo automated tests and use minimums to prevent regression

## Changes

* run coverage check as part of bamboo cicd
* fail if coverage falls below current thresholds
* coverage thresholds set to pass current code

## PR Checklist

- [ ] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: bamboo/unit-tests.sh [hunk 1]
```diff
@@ -8,4 +8,4 @@ set -ex
 docker ps -a ## Show running containers for output logs
 
 docker exec -i ${container_id}\_build_env_1 /bin/bash -c "cd $UNIT_TEST_BUILD_DIR && npm run db:local:reset"
-docker exec -i ${container_id}\_build_env_1 /bin/bash -c "cd $UNIT_TEST_BUILD_DIR && npm run test:ci"
+docker exec -i ${container_id}\_build_env_1 /bin/bash -c "cd $UNIT_TEST_BUILD_DIR && npm run test:coverage"
```
Comment (Jkovarik) [inline bamboo/unit-tests.sh:11]: We should have an arch (or team if there's a hurry)  briefing re: these changes before merging.

**Review by Jkovarik — CHANGES_REQUESTED** [review]
Requesting we not merge this before a team briefing/walkthrough.

**Comment by etcart** [comment]
> Requesting we not merge this before a team briefing/walk through.

that sounds reasonable to me, thanks

File: README.md [hunk 2]
```diff
@@ -237,6 +237,21 @@ export LOCALSTACK_HOST=127.0.0.1
...
 
+### Coverage tests
+
+If tests are working, run coverage tests
+```sh
+export LOCAL_ES_HOST=127.0.0.1
+export LOCALSTACK_HOST=127.0.0.1
+npm test:coverage
+```
+These tests will fail if coverage drops below certain thresholds.
```
Comment (Jkovarik) [inline README.md:248]: Nit: fails if coverage drops *or* the unit test fails.

**Review by Jkovarik — APPROVED** [review]
Approving as-is, with the caveat that I'm relying on CI to validate the nyc configurations pass against current main, however there are a couple of minor bits of feedback for your consideration.     Great work!

**Review by Jkovarik — APPROVED** [review]

**Pull request merged**